test/index.js

```diff
@@ -1 +1,1159 @@
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When target file does not exist => null',
+        pathObj: {
+          path: 'test/fixtures/fined',
+          extensions: ['.json', '.js'],
+        },
+        defaultObj: {
+          name: 'aaa',
+          cwd: cwd,
+        },
+        expected: null,
+      },
+    ],
+  },
+  {
+    name: 'About that 2nd argument supplements 1st argument',
+    cases: [
+      {
+        name: 'When 1st argument is full',
+        pathObj: {
+          name: 'package',
+          path: 'test/fixtures/fined',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        defaultObj: {},
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'package.json'),
+          extension: '.json',
+        },
+      },
+      {
+        name: 'When 1st argument is empty and 2nd argument is null',
+        pathObj: {},
+        defaultObj: {
+          name: 'package',
+          path: 'test/fixtures/fined',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'package.json'),
+          extension: '.json',
+        },
+      },
+      {
+        name: 'When 1st argument lacks some properties',
+        pathObj: {
+          name: 'app',
+          cwd: path.resolve(cwd, 'test'),
+        },
+        defaultObj: {
+          path: 'fixtures/fined',
+          extensions: ['.json', '.js'],
+          findUp: false,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When both 1st and 2nd arguments are full',
+        pathObj: {
+          name: 'README',
+          path: '.',
+          extensions: ['.md', '.txt'],
+          findUp: true,
+          cwd: path.resolve(cwd, 'test/fixtures'),
+        },
+        defaultObj: {
+          path: 'fixtures/fined',
+          extensions: ['.json', '.js'],
+          findUp: false,
+          name: 'app',
+          cwd: path.resolve(cwd, 'test'),
+        },
+        expected: {
+          path: path.resolve(cwd, 'README.md'),
+          extension: '.md',
+        },
+      },
+      {
+        name: 'When both 1st and 2nd arguments are empty => null',
+        pathObj: {},
+        defaultObj: {},
+        expected: null,
+      },
+      {
+        name: 'When 1st argument is a string : 1st arg. is treated as ' +
+              '{path: arg}',
+        pathObj: 'test/fixtures/fined',
+        defaultObj: {
+          name: 'app',
+          extensions: ['.json', '.js'],
+          cwd: cwd,
+          findUp: false,
+        },
+        expected: {
+          path: path.resolve(cwd, 'test/fixtures/fined', 'app.js'),
+          extension: '.js',
+        },
+      },
+      {
+        name: 'When 1st arguments is null => null',
```

Comment:
Because of usage in js-liftoff. I determined this behaviors by considering the case that `configFiles` of js-liftoff is null or its `path` property is null. But I'm OK to change this behaviors.